### DISCUSSION TITLE
UI element font size configuration

### DISCUSSION THREAD

**Discussion by chipbite** [freeplane/freeplane#345] [Ideas]
Hi! Thanks for a wonderful tool.
The tabs in the latest UI release dont scale up on my machine.
So it looks like this, the fonts on the tabs/sheets are really small:
![image](https://user-images.githubusercontent.com/77722885/166638800-36b17472-a443-4550-b2c5-8579f4a9b89e.png)

Factors / setup:
* 32" 4K screen
* 150% scaling setup
* Scaling w system is blurry, so setting app to scale - which is needed and works for e g slack, teams, etc. 

Is there maybe a setting to handle the font size of the tabs?
Using flat intellij setup, which was the default after the update. So, i e, have not messed with the defaults.

BR! /marcus

Shots:
![image](https://user-images.githubusercontent.com/77722885/166638954-9aa9d63a-3791-495b-a770-bad96cc0b8e4.png)
![image](https://user-images.githubusercontent.com/77722885/166638724-bd20d0a3-80a5-4218-b2a0-96c4a4731612.png)

**Comment by EdoFro** [comment]
hi,
look at this [post](https://github.com/freeplane/freeplane/discussions/276#discussioncomment-2441354). It may be from some help to your problem.

Hope it helps,
edo

**Reply by chipbite** [reply]
Aha, thanks, looks good, will read this!

**Reply by chipbite** [reply]
Thanks for this! 

OK, I had those settings already, mostly - so, AFAICS:

* **monitor size** seems to make no difference. 
  * The fontsize gap is the same, regardless of if I run it on the laptops screen, a 27" or a 32" screen.
  * (I have not changed the setting in the prefs, the gap is identical across screen sizes)
* `Menu item font size` enables me to scale down the menu font size, but in effect for this problem only enables me to make the menus as small and unreadable as the tab texts, haha :-D 
  * Btw, this setting was already at 14 for me, not 10 as in the example in that post. Not sure why.

So, **which setting** can I change to **_increase the font used for the labels on the tabs_**, do you know?

I will also try playing around a bit with the settings later, to see if I can randomly hit the right button.

BR! /marcus

My settings:
![image](https://user-images.githubusercontent.com/559199/167245274-e31ce804-6714-410f-9500-0867ec9d4dab.png)

**Reply by chipbite** [reply]
Hi! A bit of report back! 
**_TLDR:_** I would like to have a way to set the font size (perhaps relative to menubar font size?) of the tab labels. Is there a setting for tab label font size? Secondly, the height of the tabs are unnecessarily high, would be good to shrink them a bit.

* Updated to latest version today, **1.10.2**.
* I have played around with the settings, but I fail at getting a combination that works well.  
* Using the display size setting and keeping menu font size small, affects the relation, but not dramatically. I cannot put it more exact, I havent figured it out yet.

To make it work, a way of setting the tab label font size would be good.

So, 
* menubar font size is settable and works out well
* tab label font size does not behave the same, and I dont know how to set it
Observation:
* tabs are unnecessarily tall, even when I set menufont size to 10 and screen to small, so that they are more uniform, the tab themselves are unnecessarily tall. Not sure why.


Is there a `theme` that behaves differently w r t tabs? 
I currently have `flat intellij`.

**Comment by dpolivaev** [comment]
I don't see any bug. The font on the tabs is smaller than on the menus. It is the same as in the formatting panel items.
The setting of monitor size in the preferences allows Freeplane to correctly scale its UI elements.

**Reply by chipbite** [reply]
Hello @dpolivaev , thanks for you reply, appreciated!

> I don't see any bug. 

Aha, the fontsize gap is no bug, I assume by design then?
If so, what is the purpose of having the tab fontsize 60% smaller in relation to the menubar?

I would say the most common practice is exactly same or roughly same size (office excel, google sheets, adobe, inkscape, etc) - wouldnt you agree? 

> The font on the tabs is smaller than on the menus.

That is not the way it used to be, it seems.
Freeplane used to show the same fontsize for tabs and menus before. 
If you look at screenshots of freeplane on the internet, it is not hard to see this. 
Please find a borrowed screenshot **_below_** to verify.

Judging from this, it seems that with the latest release(s), this has changed.

> It is the same as in the formatting panel items.

Not sure what you mean by formatting panel. The toolbar?
Would be good to understand.

> The setting of monitor size in the preferences allows Freeplane to correctly scale its UI elements.

Aha! So there are 
1. parts of the UI that is scaled by freeplane, apparently the tabs - and to do so correctly, I need to set my monitor size.
2. parts in the UI (menubar) that is scaled differently, they scale ok without this setting, apparently. 

Sorry, I now maybe understand! So 
* there is not supposed to be a gap in fontsize, 
* but its not a bug, its that the user needs to configure screen size to make the scaling approaches align?

I will go and find and try this setting!
Thanks!

BR! /marcus

Shots:
1. Freeplane today, menu (green arrow), tabs (red arrow). Tab font is 60% smaller.
![image](https://user-images.githubusercontent.com/559199/167244132-19ae36e8-6b2a-4fe3-9fce-403ab65f21fe.png)

2. Freeplane previously (image from internet, https://python.iitter.com/other/119594.html )

![image](https://user-images.githubusercontent.com/559199/167244240-994d86b8-a018-43b1-90d1-24173eff21e1.png)

**Reply by chipbite** [reply]
OK, @dpolivaev - I am checking the settings to see how I can make it work, I just havent found the right setting yet,

Thanks again!

**Comment by dpolivaev** [comment]
I checked the code and the history and I can give you a complete answer now.
The logic calculating the font size on the tabs has not been changed since 2016.
It is set to 8 pt fix independently from any other UI settings but considers your monitor size to properly calculate the scaling. The smaller is the monitor size you configure the larger are all fonts.

The reason for using a pretty small font size is to occupy less space when many mind maps are open.

Nobody else has ever suggested changing this logic in the past. However, if we get other voices supporting your idea I could easily do it.

**Reply by chipbite** [reply]
> I checked the code and the history and I can give you a complete answer now.
> The logic calculating the font size on the tabs has not been changed since 2016.

That is not what I am asking. I am not debating code changes.
I simply want to get the tabs as readable as the menubar.

**Reply by dpolivaev** [reply]
It has never been configurable.

**Comment by dpolivaev** [comment]
I want to describe the question in other words and hope to get feedback from the Freeplane community.

Currently, there are many user options in section Preferences…->Appearance->Look and feel that configure the size of the UI element fonts:
 
- Monitor size (inches)
- Scale UI Fonts
- Menu item font size
- Format panel size

The first option influences the scaling of all fonts and is not limited to the UI elements.
All other options refer to different parts of the UI. And only the Monitor size option changes the size of the font on the mind map tabs used to switch the edited mind map.

Because adding a new option would make it even more confusing I could replace the last three options with a single option `UI font scaling factor` with values 60%, 70%, 80%, 90%, 100%, 110%, 120%, 130%, 150%. This factor would be applied to all UI fonts sizes. It does not mean that all sizes would be the same, I could maintain a hardcoded ratio between them, but using this option the user could influence the size of all UI elements.

Do you see how I mean it?
What do you think?

**Reply by dpolivaev** [reply]
@rexelbartolome @macmarrum @quickfold I would like to know how you see it

**Reply by macmarrum** [reply]
A good idea 👍 
As far as I know, the flat LaF themes that are available in Freeplane specify their own font: family and size.
Also people have different needs as to how big the Tool panel should be or the tab.
Therefore I would take it a step further and allow specifying the scaling factor for each area separately, rather than hard-coding the ratio.

**Reply by rexelbartolome** [reply]
Can't really say, I haven't tinkered with it since the default UI fonts are already good enough for me. But I'm up for making the font of the mindmap tabs slightly bigger than they currently are. It is, after all, a bit weird that other similar UI elements like the formatting tabs are bigger compared to the open mindmap tabs, so making making their font size the same could be a great way to make the elements feel more consistent.

![image](https://user-images.githubusercontent.com/88552647/167461078-745a8019-a55a-4dfb-9fb5-6b45896963fa.png)

Another option would be to make the `open mindmap tabs` slightly bigger but smaller than the `formatting tabs`. Then make the `formatting tabs` match the size of the `open mindmap tabs`

**Reply by quickfold** [reply]
I don't see adding more configurable options as confusing for the user, so I suggest exposing all font size UI elements in the settings to allow users to configure them. Font size is one of the most individually varying preferences based on eyesight, lighting, distance, etc. So I think it is best to remove any hidden hardcoding for this element (whether absolute font size or relative font size between elements).

**Comment by GiovanniVG** [comment]
I have the same problems as @chipbite 
With prior freeplane versions, I found a suitable setting (monitor size / scale fonts / font size / ....), but with the latest version (1.10.3) it´s impossible for me, to find a acceptable setting. Testing with "abnormal" font size settings shows that not all icons and fonts are scaled in the same way... => attached screenshot (filter icons are extremely tiny)

![fontsize](https://user-images.githubusercontent.com/109510804/179472567-377b4179-d5c8-4a8d-82bc-08a861033f90.JPG)
t

**Reply by dpolivaev** [reply]
I created an issue #505 because I can forget it otherwise.  Please upload your auto.properties so that I can check the problem.
I also recommend trying different look and feel variants.

**Reply by GiovanniVG** [reply]
I switched "look & feel (Erscheinungsbild)" from "Standard" to "Flat InteliJ" and re-changed some font settings, now everything is pretty fine for me! Obviously a problem of "Standard" preset ?
Thanks a lot! 👍🏻
